app/scripts/directives/serviceIntegration.js

```diff
@@ -0,0 +1,242 @@
+'use strict';
+
+(function() {
+  angular.module('openshiftConsole').component('serviceIntegration', {
+    controller: [
+      '$filter',
+      'APIService',
+      'AuthorizationService',
+      'BindingService',
+      'DataService',
+      'NotificationsService',
+      ServiceIntegration
+    ],
+    bindings: {
+      integration: '<',
+      consumerService: '<?'
+    },
+    templateUrl: 'views/directives/_service-integration.html'
+  });
+
+  function ServiceIntegration(
+                          $filter,
+                          APIService,
+                          AuthorizationService,
+                          BindingService,
+                          DataService,
+                          NotificationsService) {
+    var ctrl = this;
+
+    var deploymentPreferredVersion = APIService.getPreferredVersion('deployments');
+    var bindingPreferredVersion = APIService.getPreferredVersion('servicebindings');
+    var instancePreferredVersion = APIService.getPreferredVersion('serviceinstances');
+    var podPresetPreferredVersion = {group: "settings.k8s.io", version: "v1alpha1", resource: "podpresets"};
+
+    var isServiceInstanceReady = $filter('isServiceInstanceReady');
+    var isBindingReady = $filter("isBindingReady");
+
+    var integrationName = ctrl.integration.spec.externalMetadata.serviceName;
+
+    ctrl.$onInit = function() {
+      var context = {namespace: ctrl.consumerService.metadata.namespace};
+      // watch for the pod preset for this integration
+      DataService.watch(podPresetPreferredVersion, context, function(podPresets){
```

Comment:
You're leaking websockets since you're not calling `DataService.unwatch` on these watches.